### DISCUSSION TITLE
Vitest Browser mode timing out with lots of tests

### DISCUSSION THREAD

**Discussion by tylerbmyers** [vitest-dev/vitest#6536] [Feedback]
I am attempting to use Vitest workspaces to migrate tests to Vitest Browser alongside vitest-browser-react. I am seeing some weird behavior, so I wanted to start a discussion while I work on creating a reproduction (can't post my private repo here where I am seeing the issue).

Some numbers for context:

- Full test suite has 130 test files and 1481 tests
- The browser workspace has 4 test files and 57 tests.
- I have increased the poll and test timeout to 10000 seconds because I have some tests that do api retries that take a bit of time in the browser.

Some things I am noticing:

- My tests run as the full workspace pass every single time when I run in watch mode
- My browser project passes every time in both watch mode and run mode
- The same test suite run as a full workspace fails (the tests that fail vary in quantity but are more or less the same ones) when I run them in run mode
- The tests that fail seem to be failing because MSW (used to mock API calls) is taking a really long time to respond (sometimes up to 30 seconds)
- I cannot confirm this is true deterministically but I have noticed that when I have an `expect.element` right after rendering the component MSW takes a really long time to run. If I do something like click or hover on an element instead, it returns almost instantly. This leads me to wonder if something with vitest's implementation of module mocking with MSW is interfering with the api mocks. I have not been able to dig into it too deeply yet, but wanted to flag this observation.

I will update this discussion with more details and a reproduction if I am able to create one.

Thanks!

**Comment by sheremet-va** [comment]
> The tests that fail seem to be failing because MSW (used to mock API calls) is taking a really long time to respond (sometimes up to 30 seconds)

Is it your own MSW instance?

**Reply by tylerbmyers** [reply]
Yes it is. Is the one vitest is using exposed?

**Reply by sheremet-va** [reply]
It is not, but I wanted to experiment with exposing it somehow or redirecting the `msw` import to our own instance

`msw` recommends having a single `msw` instance if code runs in iframes, so it doesn't need to restart in every iframe

**Reply by tylerbmyers** [reply]
That would make sense to me since it’s already being used for the other mocking. Happy to help test that out at some point in the branch I’m working on in my repo.

**Comment by tsirlucas** [comment]
looks related to #7139
